spec/lib/solidus_kustomer/client_spec.rb

```diff
@@ -76,14 +76,52 @@
     end
   end
 
+  describe '#find_customer' do
+    context 'when the Spree::User has a kustomer_id' do
+      it 'retrieves the customer via their UUID' do
+        kustomer = described_class.new(api_key: 'my_api_key')
+        user = create(:user, kustomer_id: 'the_customer_UUID')
+        allow(kustomer).to receive(:find_customer_by_uuid)
```

Comment:
You never stub the subject under test. The fact that the class relies on `find_customer_by_uuid` and the other methods is an internal detail. What you should stub here are the external interactions of the method.

The same goes for the other tests.